### PR TITLE
Propose Guidelines for Creating and Executing Tests in Vanilla OS

### PR DESCRIPTION
**Pull request by mirkobrombin**

### CODE REVIEW

**Review by pietrodicaprio — APPROVED** [review]

**Review by axtloss — APPROVED** [review]

**Review by taukakao — APPROVED** [review]

File: TESTS_GUIDELINES.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+- Detailed instructions on how to execute the test
+- Acceptance criteria for passing the test
+
+The documentation can be provided as a comment in the test itself, making it easily accessible to everyone in the most expected place.
+
+## Human Testing
+
+While automated tests are crucial for validating the functionality and security of the project, they cannot cover every aspect of the user experience. It is essential to conduct manual tests, especially for new features and critical fixes, to ensure they meet the expected standards in real-world scenarios.
+
+Manual testing includes usability tests, user interaction tests, and comprehensive scenario tests. It is crucial not to rely soley on automated tests but also involve human insights to catch issues that automated tests may miss.
```
Comment (kbdharun) [inline TESTS_GUIDELINES.md]:
```suggestion
Manual testing includes usability tests, user interaction tests, and comprehensive scenario tests. It is crucial not to rely solely on automated tests, but also involve human insights to catch issues that automated tests may miss.
```

**Review by kbdharun — APPROVED** [review]
LGTM, after this minor suggestion.

File: TESTS_GUIDELINES.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# Guidelines for Creating and Executing Tests in Vanilla OS
+
+## Introduction
+
+This document serves as a reference for maintainers of Vanilla OS repositories in creating and managing testing procedures. The goal is to ensure the integrity, security, and functionality of components through consistent and well-structured tests, identifying potential issues before they are brought into production.
+
+It is important that all maintainers are familiar with and have understood the [Vanilla OS contribution guidelines](https://github.com/Vanilla-OS/.github/blob/main/CONTRIBUTING.md).
```
Comment (TheEvilSkeleton) [inline TESTS_GUIDELINES.md]:
```suggestion
It is important that all maintainers are familiar with and have understood the [Vanilla OS contribution guidelines](CONTRIBUTING.md).
```

**Review by taukakao — APPROVED** [review]

File: TESTS_GUIDELINES.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+## Responsibilities of Maintainers
+
+Maintainers are responsible for the continuous integration of tests into their projects. Each new feature or fix should be accompanied, where possible, by relevant tests that validate the changes and ensure that existing functionalities remain intact.
```
Comment (TheEvilSkeleton) [inline TESTS_GUIDELINES.md]: What about "It is the responsibility of the maintainers to ensure the continuous integration (CI) of tests into their respective projects."?

File: TESTS_GUIDELINES.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+Each test case must strictly correspond to the functional requirements of the component. Clearly define what is being tested and what result is expected.
+
+### Implementation of Tests
+
+For the implementation of tests in projects written in Go, the language predominantly used for backend/critical components of the project, Vanilla OS adopts the official Go testing suite. For other languages, any other suite can be used, considering ease of use and portability, to emphasize its use.
+
+### Automation and Integration of Tests
+
+Integrate the tests into the Continuous Integration (CI) process using GitHub Actions to automate the execution of tests with every push or pull request. This integration ensures that tests are systematically executed, providing immediate feedback on the code's status and preventing the merging of changes that do not pass the tests.
```
Comment (TheEvilSkeleton) [inline TESTS_GUIDELINES.md]: > Continuous Integration (CI)

If the above suggestion is accepted, we could just use "CI"

**Review by TheEvilSkeleton — CHANGES_REQUESTED** [review]

File: TESTS_GUIDELINES.md (hunk 3 above)
Comment (mirkobrombin) [reply TESTS_GUIDELINES.md]: Yeah I like your proposal

**Review by 89luca89 — APPROVED** [review]

**Review by TheEvilSkeleton — APPROVED** [review]

File: TESTS_GUIDELINES.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+### Automation and Integration of Tests
+
+Integrate the tests into the CI process using GitHub Actions to automate the execution of tests with every push or pull request. This integration ensures that tests are systematically executed, providing immediate feedback on the code's status and preventing the merging of changes that do not pass the tests.
+
+## Quality Standards
+
+### Clean Code
+
+Maintain clean, simple and targeted test code.
```
Comment (matbme) [inline TESTS_GUIDELINES.md:31]: Not sure if this is the right place, but we should enforce formatters like gofmt (integrated into gopls) for Go and black (or ruff which checks for black's rules) for Python.

Comment (mirkobrombin) [reply TESTS_GUIDELINES.md:31]: Not sure if this should be in those guidelines or the general ones

Comment (matbme) [reply TESTS_GUIDELINES.md:31]: Maybe it's best to put in the general guidelines.

**Review by matbme — APPROVED** [review]

**Pull request merged**